ci.md

```diff
@@ -0,0 +1,201 @@
+    * Builds are run inside Docker containers for Bionic or Focal.
+    * By default, dependencies are installed from stable debian binaries for
+      stable branches, and nightlies for `main` branches.
+    * Library being tested is compiled using CMake and Make.
+* `<library>-ci-pr_any-homebrew-amd64`: Build and test on macOS
+    * Dependencies are installed from homebrew.
+    * Library being tested is compiled using CMake and Make.
+* `<library>-ci-pr_any-windows*-amd64`: Build and test on Windows
+    * External dependencies are installed from vcpkg
+    * Library being tested and its Ignition dependencies are compiled using colcon.
+
+Pull requests to stable branches also trigger:
+
+* `<library>-abichecker-any_to_any-ubuntu_auto-amd64`: Run ABI checker
+
+### GitHub Actions
+
+GitHub actions run on GitHub's servers. Currently, it builds Linux for all
+libraries, and macOS for some of them.
+
+Every commit to any branch will trigger:
+
+* `Ubuntu CI / Ubuntu Bionic CI (push)`: Build, test and run linters on Ubuntu Bionic.
+    * Builds are run inside Docker containers.
+    * By default, dependencies are installed from stable debian binaries for
+      stable branches, and nightlies for `main` branches.
+    * Library being tested is compiled using CMake and Make.
+    * Test coverage is computed and uploaded to Codecov.
+* `Ubuntu CI / Ubuntu Focal CI (push)`: Build, test and run linters on Ubuntu Focal.
+    * Builds are run inside Docker containers.
+    * By default, dependencies are installed from stable debian binaries for
+      stable branches, and nightlies for `main` branches.
+    * Library being tested is compiled using CMake and Make.
+
+Every pull request will also trigger the exact same builds, but with the
+`pull_request` suffix. Note that pull requests from forks won't have the
+`push` checks, while pull requests from the official repositories will have
+both `pull_request` and `push`.
+
+Other checks run on GitHub Actions:
+
+* `DCO`: Checks that all commits are signed correctly for the
+    [Developer Certificate of Origin](https://developercertificate.org/).
+* `PR Collection Labeler`: Adds collection labels (i.e. Blueprint, Citadel...)
+    to the pull request automatically according to the target branch.
+* `Ticket opened`: Adds the pull request to the [Core development board](https://github.com/orgs/ignitionrobotics/projects/3).
+* `codecov/*`: Checks that the test coverage hasn't been reduced.
+
+## Required checks
+
+Some checks are marked as `Required`. This means the pull request can't
+be merged unless that check is green.
+
+Ideally, all checks would be required. But known warnings, test failures and
+flaky tests often prevent us from marking them as required. Until these are
+addressed, some checks will need to be checked by hand when they fail, to see
+if those failures are unexpected. See
+[this issue](https://github.com/ignition-tooling/release-tools/issues/398) for
```

Comment:
Could there be a one-sentence description in that issue saying, the checked things are required? That wasn't obvious from the title, with "check" meaning different things in "CI checks" and "All checked" in the checkbox lists. A bit of a mind pause there.

Reply:
I added a few sentences :sweat_smile: Once this guide is live, we can link to it from that issue for better context.